application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,799 @@
+    <property>
+      <code>define('mostViewedPagesTranslation', {
+  prefix: 'Analytics.Code.Macros.',
+  keys: [
+    "label",
+    "nb_visits",
+    "nb_hits",
+    "sum_time_spent",
+    "entry_nb_visits",
+    "entry_nb_actions",
+    "bounce_rate",
+    "exit_rate"
+  ]
+});
+
+
+require(['jquery', 'xwiki-l10n!mostViewedPagesTranslation'], function ($, l10n) {
+  function getAttributes(element) {
+    return Array.from(element.attributes).reduce((acc, attribute) =&gt; {
+      acc[attribute.name] = attribute.value;
+      return acc;
+    }, {});
+  }
+  function constructUrl(attributes) {
+    const paramString = $.param({
+      module: 'API',
+      method: 'Actions.getPageTitles',
+      idSite: attributes.id,
+      period: 'range',
+      date: `${attributes.startdate.replaceAll('/', '-')},${attributes.enddate.replaceAll('/', '-')}`, //the data processing could be moved to velocity
+      format: 'json',
+      filter_sort_column: 'nb_hits',
+      filter_sort_order: 'desc',
+      token_auth: attributes.auth
+    });
+    return attributes.ip + 'index.php?' + paramString;
+  }
+
+
+  function createHeader(headersTranslations) {
+    let headerRow = document.createElement('tr');
+    headersTranslations.forEach(header =&gt; {
+      let headerCell = document.createElement('th');
+      headerCell.classList.add('xwiki-mostviewedpages-header');
+      headerCell.textContent = l10n[header];
+      headerRow.appendChild(headerCell);
+    });
+    return headerRow;
+  }
+  function createTable(table, attributes) {
+    let fragment = document.createDocumentFragment();
+    table.classList.add('xwiki-mostviewedpages-table');
+
+    let url = constructUrl(attributes);
+    console.log(url);
+    $.getJSON(url)
+      .done(function (data) {
+        let headers = ['label', 'nb_visits', 'nb_hits', 'sum_time_spent', 'entry_nb_visits', 'entry_nb_actions', 'bounce_rate', 'exit_rate'];
+
+        fragment.appendChild(createHeader(headers));
+
+        data.forEach((item, index) =&gt; {
+          let row = document.createElement('tr');
+          if (index % 2 === 0) {
+            row.classList.add('xwiki-mostviewedpages-evenRow');
+          }
+          headers.forEach(header =&gt; {
+            let cell = document.createElement('td');
+            cell.classList.add('xwiki-mostviewedpages-cell');
+            cell.textContent = item[header];
+            row.appendChild(cell);
+          });
+          fragment.appendChild(row);
+        });
+        table.innerHTML = '';
+        table.appendChild(fragment);
+      })
+      .fail(function (jqxhr, textStatus, error) {
+        let err = textStatus + ", " + error;
+        console.log("Request Failed: " + err);
+        // Handle the error in a user-friendly way here.
+      });
+  }
+
+  $(function () {
+    let tables = document.querySelectorAll('.mostViewedPages');
+    console.log(tables)
+    tables.forEach((table, index) =&gt; {
+      let attributes = getAttributes(table);
```

Comment:
You don't need this
Inside the constructURL, the data attributes can be accessed directly for your table